javascript/swiggy.js

```diff
@@ -0,0 +1,158 @@
+                "imgUrl": "https://media.istockphoto.com/photos/fish-biryani-with-basmati-rice-indian-food-picture-id488481490?k=20&m=488481490&s=612x612&w=0&h=HYP2KxiC1e2tAtzmfrA7xxs3u8LD1wjSLPUD9bZ48eU=",
+                "categories": ["recommended", "biryani"]
+            },
+            {
+                "id": "2122",
+                "displayName": "Real Biryani",
+                "price": 349,
+                "currency": "INR",
+                "vegetarian": false,
+                "imgUrl": "https://media.istockphoto.com/photos/hyderabadi-biryani-a-popular-chicken-or-mutton-rice-preparation-picture-id466089615?k=20&m=466089615&s=612x612&w=0&h=_Z0Jlombq-VX8Pl8I9mJf_kIuvbzZ7j8ucxvRoLL8BM=",
+                "categories": ["biryani"]
+            },
+            {
+                "id": "2123",
+                "displayName": "Plain Veg Biryani",
+                "price": 149,
+                "currency": "INR",
+                "vegetarian": true,
+                "imgUrl": "https://media.istockphoto.com/photos/chicken-biryani-directly-above-photo-picture-id1169141170?k=20&m=1169141170&s=612x612&w=0&h=EpCF3lQF2GBRaVApNELuE5xFQfv8fyQ_wWC52hmyxeo=",
+                "categories": ["biryani"]
+            }
+        ];
+    }
+
+    const menuItems = fetchMenuItems();
+
+    function fetchCartItems() {
+        return {
+            "lineItems": [{
+                "id": "1121",
+                "name": "Plain Veg Biryani",
+                "quantity": 2,
+                "price": 149,
+                "currency": "INR"
+            }],
+            "shippingFee": 0,
+            "discount": 0,
+            "tax": 0,
+            "subTotal": 149
+        }
+    }
+
+    const cartItems = fetchCartItems();
+
+    const itemsByCategoryMap = new Map();
+
+	function addItemByCategoryInMap(key,element){
+		if (!itemsByCategoryMap.has(key)) {
+			itemsByCategoryMap.set(key, []);
+		}
+		itemsByCategoryMap.get(key).push(element);
+	}
+	function makeItemsByCategoryMap(element){
+			element.categories.forEach(key => addItemByCategoryInMap(key,element));
+	}
+	
+    menuItems.forEach(element => makeItemsByCategoryMap(element));
+
+    const main = document.querySelector("main");
+
+    const categoriesContainer = document.querySelector(".categories");
+
+    const categoriesList = document.createElement("ul");
+	function addListItemInCategoryList(element){
+        const categoryName = document.createElement("li");
+        categoryName.id = element.id;
+        categoryName.innerText = element.displayName;
+        categoriesList.append(categoryName);
+    }
+    categories.forEach(element => addListItemInCategoryList(element));
+    categoriesContainer.append(categoriesList);
+
+    const menuItemsContainer = document.querySelector(".menu-items");
+
+    function addCategoryInfo(key) {
+        return `<h3>${key}</h3>
+		<p>${itemsByCategoryMap.get(key).length} Items</p>`
+    }
+
+    function addItemDescription(key, index) {
```

Comment:
You are not adding description inside this function - this function is simply creating a template and should be called createItemTemplate - it should accept `itemsByCategoryMap` and all the required parameters and return the template.

This should just be a pure function - you don't need to use closure here.